### PR TITLE
Capability changing to status predicate

### PR DESCRIPTION
**Pull request by zahnno**
Rule predicate to make things easier when looking for a ucj capability status transistion.

Similar to IsCapabilityOfCertainCategoryAndStatus except predicate checks specified capability.

### CODE REVIEW

File: src/foam/nanos/crunch/predicate/CapabilityJunctionTransistionToStatus.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.predicate',
+  name: 'CapabilityJunctionTransistionToStatus',
```
Comment (jlhughes) [inline src/foam/nanos/crunch/predicate/CapabilityJunctionTransistionToStatus.js]:
```suggestion
  name: 'CapabilityJunctionTransitionToStatus',
```

File: src/foam/nanos/crunch/predicate/CapabilityJunctionTransistionToStatus.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  implements: ['foam.core.Serializable'],
+  
+  documentation: `Returns true if the capability of the ucj submitted is transistioning to status defined.`,
```
Comment (jlhughes) [inline src/foam/nanos/crunch/predicate/CapabilityJunctionTransistionToStatus.js]:
```suggestion
  documentation: `Returns true if the capability of the ucj submitted is transitioning to status defined.`,
```

File: src/foam/nanos/crunch/predicate/CapabilityJunctionTransistionToStatus.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      class: 'Reference',
+      name: 'capabilityId',
+      of: 'foam.nanos.crunch.Capability',
+      documentation: `Used to catch any ucj's with their targetId equaling this value.`
+    },
+    {
+      name: 'status',
+      class: 'Enum',
+      of: 'foam.nanos.crunch.CapabilityJunctionStatus',
+      documentation: `Status to check compare to the new ucj's status. (ie. ucj's transistion to status provided.)`,
```
Comment (jlhughes) [inline src/foam/nanos/crunch/predicate/CapabilityJunctionTransistionToStatus.js]:
```suggestion
      documentation: `Status to check compare to the new ucj's status. (ie. ucj's transition to status provided.)`,
```

File: src/foam/nanos/nanos.js [hunk 4]
```diff
@@ -484,6 +484,7 @@ FOAM_FILES([
   { name: 'foam/nanos/crunch/predicate/CapabilityPrerequisitesGranted' },
   { name: 'foam/nanos/crunch/predicate/StatusChangedTo' },
   { name: 'foam/nanos/crunch/predicate/IsAgent' },
+  { name: 'foam/nanos/crunch/predicate/CapabilityJunctionTransistionToStatus' },
```
Comment (jlhughes) [inline src/foam/nanos/nanos.js]:
```suggestion
  { name: 'foam/nanos/crunch/predicate/CapabilityJunctionTransitionToStatus' },
```

File: tools/classes.js [hunk 5]
```diff
@@ -845,6 +845,7 @@ var classes = [
   'foam.nanos.crunch.predicate.CapabilityPrerequisitesGranted',
   'foam.nanos.crunch.predicate.StatusChangedTo',
   'foam.nanos.crunch.predicate.IsAgent',
+  'foam.nanos.crunch.predicate.CapabilityJunctionTransistionToStatus',
```
Comment (jlhughes) [inline tools/classes.js]:
```suggestion
  'foam.nanos.crunch.predicate.CapabilityJunctionTransitionToStatus',
```

File: src/foam/nanos/crunch/predicate/CapabilityJunctionTransitionToStatus.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/predicate/CapabilityJunctionTransitionToStatus.js]: 2021

File: src/foam/nanos/crunch/predicate/CapabilityJunctionTransitionToStatus.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  methods: [
+    {
+      name: 'f',
+      javaCode: `
+        X x = (X) obj;
+        UserCapabilityJunction old = (UserCapabilityJunction) x.get("OLD");
+        UserCapabilityJunction ucj = (UserCapabilityJunction) x.get("NEW");
+
+        return old != null &&
+            ! old.getStatus().equals(ucj.getStatus()) &&
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/predicate/CapabilityJunctionTransitionToStatus.js]: == works for Enum's (and is faster)

File: src/foam/nanos/crunch/predicate/CapabilityJunctionTransitionToStatus.js (hunk 6 above)
Comment (zahnno) [reply src/foam/nanos/crunch/predicate/CapabilityJunctionTransitionToStatus.js]: the future is now huh

**Pull request merged**